include/unifex/any_unique.hpp

```diff
@@ -186,11 +192,22 @@ struct _with_type_erased_tag_invoke<
   struct type {
     friend Ret tag_invoke(_overload::base_cpo_t<CPO> cpo, replace_this_t<Args, Derived>... args)
         noexcept(NoExcept) {
+      using cpo_t = _overload::base_cpo_t<CPO>;
+      static_assert(
+          !NoExcept || noexcept(extract_this<Args...>{}((decltype(args) &&) args...)));
       auto&& t = extract_this<Args...>{}((decltype(args) &&) args...);
+      static_assert(!NoExcept || noexcept(get_object_address(t)));
       void* objPtr = get_object_address(t);
+      static_assert(!NoExcept || noexcept(get_vtable(t)->template get<CPO>()));
       auto* fnPtr = get_vtable(t)->template get<CPO>();
+      static_assert(!NoExcept || noexcept(fnPtr(
+          (cpo_t) (cpo_t&&) cpo,
+          replace_this<Args>::get((decltype(args) &&) args, objPtr)...)));
+      // The strange cast of the cpo to save the compiler from having to reserve
+      // space in the image for the cpo in case we pass it by reference through
+      // this function pointer:
       return fnPtr(
-          std::move(cpo),
+          (cpo_t) (cpo_t&&) cpo,
           replace_this<Args>::get((decltype(args) &&) args, objPtr)...);
```

Comment:
Are there places where we are passing the CPO by reference?
If so, we should change them to passing by-value.

I try to always pass CPO objects by-value precisely to avoid ODR-using the CPO object.

Reply:
There are no such places currently. I experimented with passing it by reference to see its impact on generated code. Since `operator()` ostensibly has an extra hidden `this` argument, it seemed to me that passing the CPO by reference might make it possible for the dispatch function into a simple tailcall. But really, the tailcall doesn't really matter because the compiler generally inlines the application of the CPO anyway. I can remove this.

Reply:
I generally thought compilers did a pretty good job of completely eliminating overhead of passing empty objects by-value. e.g. https://godbolt.org/z/bK9hqs

If you pass an empty object by-reference then you end up consuming an extra register to pass the address through.

Reply:
You misunderstand the point of my experiment. If the only thing a function (A) does is call another function (B) with the same arguments, then the compiler rewrites A as a `jmp` to B. My experiment here was in trying to jigger things so that the thunk gets turned into a `jmp` by putting the same arguments on the stack that will be used inside the thunk to do the dispatch. I hoped to save space that way.

That's not what was happening, so I can remove this part of the experiment.